File: 258/add-digits.js
/**
 * @param {number} num
 * @return {number}
 */
var addDigits = function (num) {
    while (num.toString().length > 1) {
        num = num.toString().split('').map(item => item - '0').reduce((acc, x) => acc + x);
    }
    return num;
};

File: 258/AddDigits.scala
object AddDigits {
  def addDigits(_num: Int): Int = {
    var num = _num
    while (num >= 10) {
      var sum = 0
      while (num > 0) {
        sum += num % 10
        num = num / 10
      }
      num = sum
    }

    num
  }

  def main(args: Array[String]): Unit = {
    val num = 38
    println(addDigits(num))
  }
}
